src/main/java/com/brave/playlist/fragment/AllPlaylistFragment.kt

```diff
@@ -0,0 +1,144 @@
+    private lateinit var mRvRecentlyPlayed: RecyclerView
+    private lateinit var mRvPlaylist: RecyclerView
+    private lateinit var mTvRecentlyPlayed: TextView
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        mPlaylistViewModel = ViewModelProvider(requireActivity())[PlaylistViewModel::class.java]
+
+        mPlaylistToolbar = view.findViewById(R.id.playlistToolbar)
+
+        mBtAddNewPlaylist = view.findViewById(R.id.btAddNewPlaylist)
+        mBtAddNewPlaylist.setOnClickListener {
+            val newPlaylistFragment = NewPlaylistFragment.newInstance(
+                PlaylistOptionsEnum.NEW_PLAYLIST
+            )
+            parentFragmentManager
+                .beginTransaction()
+                .replace(android.R.id.content, newPlaylistFragment)
+                .addToBackStack(AllPlaylistFragment::class.simpleName)
+                .commit()
+        }
+        mRvRecentlyPlayed = view.findViewById(R.id.rvRecentlyPlayed)
+        mRvPlaylist = view.findViewById(R.id.rvPlaylists)
+
+        mTvRecentlyPlayed = view.findViewById(R.id.tvRecentlyPlayed)
+
+        mPlaylistViewModel.fetchPlaylistData(ConstantUtils.ALL_PLAYLIST)
+
+        mPlaylistViewModel.allPlaylistData.observe(viewLifecycleOwner) { allPlaylistData ->
+            Log.e(TAG, allPlaylistData.toString())
+            val allPlaylistList = mutableListOf<PlaylistModel>()
+
+            var defaultPlaylistModel: PlaylistModel? = null
+            for (allPlaylistModel in allPlaylistData) {
+                val playlistModel = PlaylistModel(
+                    allPlaylistModel.id,
+                    allPlaylistModel.name,
+                    allPlaylistModel.items
+                )
+
+                if (playlistModel.id == DEFAULT_PLAYLIST) {
+                    defaultPlaylistModel = playlistModel
+                } else {
+                    allPlaylistList.add(
+                        playlistModel
+                    )
+                }
+            }
+            defaultPlaylistModel?.let { allPlaylistList.add(0, it) }
+
+            val recentPlaylistJson =
+                PlaylistPreferenceUtils.defaultPrefs(requireContext()).recentlyPlayedPlaylist
+            if (!recentPlaylistJson.isNullOrEmpty()) {
+                val recentPlaylist = LinkedList<PlaylistModel>()
+                val recentPlaylistIds: LinkedList<String> = GsonBuilder().create().fromJson(
+                    recentPlaylistJson,
+                    TypeToken.getParameterized(LinkedList::class.java, String::class.java).type
+                )
+                if (recentPlaylistIds.size > 0) {
+                    recentPlaylistIds.forEach ids@{
+                        allPlaylistList.forEach models@{ model ->
+                            if (model.id == it && model.items.isNotEmpty()) {
+                                recentPlaylist.add(model)
```

Comment:
1. This list conversion can be done in repository 
2.  that converted list get in Viewmodel in a coroutine scope,
3. that will set to livedata
4. Fragment can observe that livedata